File: app/src/test/java/com/huynd/skyobserver/adapters/ListViewPriceOneDayAdapterTest.java
package com.huynd.skyobserver.adapters;

import android.view.View;

import com.huynd.skyobserver.activities.MainActivity;

import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.Robolectric;
import org.robolectric.RobolectricTestRunner;

import static junit.framework.Assert.assertEquals;

/**
 * Created by HuyND on 9/10/2017.
 */

@RunWith(RobolectricTestRunner.class)
public class ListViewPriceOneDayAdapterTest {
    private MainActivity mActivity;

    @Before
    public void setUp() {
        mActivity = Robolectric.setupActivity(MainActivity.class);
    }

    @Test
    public void testGetViewWithNullItem() throws Exception {
        ListViewPriceOneDayAdapter adapter = new ListViewPriceOneDayAdapter(mActivity);
        adapter.add(null);
        View view = adapter.getView(0, null, null);
        ListViewPriceOneDayAdapter.ViewHolder vH = (ListViewPriceOneDayAdapter.ViewHolder) view.getTag();
        assertEquals("", vH.getTextViewDepartTime().getText().toString());
        assertEquals("", vH.getTextViewArriveTime().getText().toString());
    }
}

File: app/src/test/kotlin/com/huynd/skyobserver/adapters/ListViewPriceOneDayAdapterTest.kt
package com.huynd.skyobserver.adapters

import com.huynd.skyobserver.activities.MainActivity
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import org.robolectric.Robolectric
import org.robolectric.RobolectricTestRunner
import kotlin.test.assertEquals

/**
 * Created by HuyND on 9/12/2017.
 */

@RunWith(RobolectricTestRunner::class)
class ListViewPriceOneDayAdapterTest {
    private lateinit var mActivity: MainActivity

    @Before
    @Throws(Exception::class)
    fun setUp() {
        mActivity = Robolectric.setupActivity(MainActivity::class.java)
    }

    @Test
    @Throws(Exception::class)
    fun testGetViewWithNullItem() {
        val adapter = ListViewPriceOneDayAdapter(mActivity)
        adapter.add(null)
        val view = adapter.getView(0, null, null)
        val vH = view.tag as ListViewPriceOneDayAdapter.ViewHolder
        vH.run {
            assertEquals("", textViewDepartTime.text)
            assertEquals("", textViewArriveTime.text)
        }
    }
}
